builtin/logical/ssh/path_issue_sign.go

```diff
@@ -150,23 +174,12 @@ func (b *backend) calculateValidPrincipals(data *framework.FieldData, req *logic
 	var allowedPrincipals []string
 	for _, principal := range strutil.RemoveDuplicates(strutil.ParseStringSlice(principalsAllowedByRole, ","), false) {
 		if role.AllowedUsersTemplate {
-			// Look for templating markers {{ .* }}
-			matched := containsTemplateRegex.MatchString(principal)
-			if matched {
-				if req.EntityID != "" {
-					// Retrieve principal based on template + entityID from request.
-					templatePrincipal, err := framework.PopulateIdentityTemplate(principal, req.EntityID, b.System())
-					if err == nil {
-						// Template returned a principal
-						allowedPrincipals = append(allowedPrincipals, templatePrincipal)
-					} else {
-						return nil, fmt.Errorf("template '%s' could not be rendered -> %s", principal, err)
-					}
-				}
-			} else {
-				// Static principal or err template
-				allowedPrincipals = append(allowedPrincipals, principal)
+			rendered, err := b.renderPrincipal(principal, req)
+			if err != nil {
+				return nil, fmt.Errorf("template '%s' could not be rendered -> %s", principal, err)
```

Comment:
This is inconsistent with above (lines 77) -- I believe the err should be returned here and this message should be preserved as-is as we already contain a copy of the error message prefix you have here (so it'll be duplicated). 

IOW, I think 77 is wrong too (`return nil, err`) to preserve existing behavior-ish.

Reply:
Oops good catch @cipherboy. I'll merge this PR and address this in another myself.

Reply:
previously (when only the allowed_users field could be templated) the template rendering happened in `calculateValidPrincipals` and the error on this line (`return nil, fmt.Errorf("template '%s' could not be rendered -> %s", principal, err)`, on line 163 in the previous version) could be returned by that func, 

That return value would then be caught on the previous version's line 74 which contained the logical.ErrorResponse wrapping, i.e. lines 73-76 prior to my changes were:
```
parsedPrincipals, err = b.calculateValidPrincipals(data, req, role, role.DefaultUser, role.AllowedUsers, strutil.StrListContains)
if err != nil {
	return logical.ErrorResponse(err.Error()), nil
} 
```

I was aiming to match the failure behaviors for templating errors in `default_user` to the existing behavior for `allowed_users` -- that all said, :+1: with any changes to it, y'all know the details on that better than me!